packages/core/src/state-management/state-manager.ts

```diff
@@ -83,11 +83,21 @@ export class StateManager {
         concatMap((tip) => this._handleTip(state$, tip))
       )
       .toPromise()
-      .then(() => {
-        if (pinned) {
-          this.syncedPinnedStreams.add(state$.id.toString())
-        }
-      })
+    if (pinned) {
+      this.syncedPinnedStreams.add(state$.id.toString())
+    }
+  }
+
+  /**
+   * If it is a lone genesis, verify the signature.
+   * @param state$
+   */
+  async verifyLoneGenesis(state$: RunningState): Promise<RunningState> {
+    if (state$.value.log.length > 1) {
```

Comment:
what if when the stream is loaded it has multiple signed commits but still no anchor commits? Will we ever verify the genesis?

EDIT: Nevermind, I see now how this is handled in `ConflictResolution.applyLogToState`.

Phew, all of this is so subtle 😬